source/server/configuration_impl.cc

```diff
@@ -171,7 +172,30 @@ WatchdogImpl::WatchdogImpl(const envoy::config::bootstrap::v3::Watchdog& watchdo
   multikill_timeout_ =
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, multikill_timeout, 0));
   multikill_threshold_ = PROTOBUF_PERCENT_TO_DOUBLE_OR_DEFAULT(watchdog, multikill_threshold, 0.0);
-  actions_ = watchdog.actions();
+  auto actions = watchdog.actions();
+
+// Add abort_action if it's available on the given platform and killing is
+// enabled since it's more helpful than the default kill action.
+#ifndef WIN32
+  envoy::extensions::watchdog::abort_action::v3alpha::AbortActionConfig abort_config;
```

Comment:
Question for the reviewers: There's some tension here given that the action `AbortAction` extension and folks can choose to omit it, but if they do this might break and they'd need to patch it.

Thoughts?